src/ResultWriter/WaveFieldWriter.h

```diff
@@ -248,6 +258,33 @@ class WaveFieldWriter : private async::Module<WaveFieldWriterExecutor, WaveField
 			}
 		}
 
+		if (m_integrals) {
+			unsigned int offset = 0;
+			unsigned int flagOffset = WaveFieldWriterExecutor::NUM_LOWVARIABLES;
+		if (m_pstrain) {
+				offset = WaveFieldWriterExecutor::NUM_LOWVARIABLES;
+				flagOffset = 0;
+			}
```

Comment:
This is a bit difficult to understand. It is not clear what offset, flagOffset, nextId and i are for. You do not need `flagOffset` (see next comment) and `offset` can be set between line 244 and 259. Saves one additional if statement.